From a7e061079e4c38fa6486488556211629592c12e9 Mon Sep 17 00:00:00 2001
From: Roman Birg <roman@cyngn.com>
Date: Thu, 19 Mar 2015 14:32:33 -0700
Subject: [PATCH 18/18] SystemUI: small optimization for notification icon
 numbers

Don't create the paint object used to draw notification icon count
badges in the status bar unless the user actually wants to see those
counts.

Change-Id: Iabba316102583a798acdc124d9fb51c0d7826a0d
Signed-off-by: Roman Birg <roman@cyngn.com>
---
 .../src/com/android/systemui/statusbar/StatusBarIconView.java    | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarIconView.java b/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarIconView.java
index 907dec13001..b9eb7c16328 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarIconView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarIconView.java
@@ -155,10 +155,6 @@ public class StatusBarIconView extends AnimatedImageView {
         mDozer = new NotificationIconDozeHelper(context);
         mBlocked = blocked;
         mSlot = slot;
-        mNumberPain = new Paint();
-        mNumberPain.setTextAlign(Paint.Align.CENTER);
-        mNumberPain.setColor(context.getColor(R.drawable.notification_number_text_color));
-        mNumberPain.setAntiAlias(true);
         setNotification(sbn);
         maybeUpdateIconScaleDimens();
         setScaleType(ScaleType.CENTER);
@@ -292,6 +288,10 @@ public class StatusBarIconView extends AnimatedImageView {
             if (icon.number > 0 && getContext().getResources().getBoolean(
                         R.bool.config_statusBarShowNumber)) {
                 if (mNumberBackground == null) {
+                    mNumberPain = new Paint();
+                    mNumberPain.setTextAlign(Paint.Align.CENTER);
+                    mNumberPain.setColor(getContext().getColor(R.drawable.notification_number_text_color));
+                    mNumberPain.setAntiAlias(true);
                     mNumberBackground = getContext().getResources().getDrawable(
                             R.drawable.ic_notification_overlay);
                 }
@@ -299,6 +299,7 @@ public class StatusBarIconView extends AnimatedImageView {
             } else {
                 mNumberBackground = null;
                 mNumberText = null;
+                mNumberPain = null;
             }
             invalidate();
         }
-- 
2.11.0

